private/rules/pom_file.bzl

```diff
@@ -13,19 +13,30 @@ _TYPED_DEP = """        <dependency>
             <type>{3}</type>
         </dependency>"""
 
+_SCOPED_DEP = """        <dependency>
+            <groupId>{0}</groupId>
+            <artifactId>{1}</artifactId>
+            <version>{2}</version>
+            <type>{3}</type>
+            <scope>{4}</scope>
+        </dependency>"""
+
 def _explode_coordinates(coords):
     """Takes a maven coordinate and explodes it into a tuple of
-    (groupId, artifactId, version, type)
+    (groupId, artifactId, version, type, scope)
     """
     if not coords:
         return None
 
     parts = coords.split(":")
     if len(parts) == 3:
-        return (parts[0], parts[1], parts[2], "jar")
+        return (parts[0], parts[1], parts[2], "jar", "compile")
     if len(parts) == 4:
         # Assume a buildr coordinate: groupId:artifactId:type:version
-        return (parts[0], parts[1], parts[3], parts[2])
+        return (parts[0], parts[1], parts[3], parts[2], "compile")
+    if len(parts) == 5:
+        # Assume a buildr coordinate: groupId:artifactId:type:scope:version
+        return (parts[0], parts[1], parts[4], parts[2], parts[3])
```

Comment:
The comment says we're returning `group:artifact:type:scope:version`, but the actual tuple returned is `group:artifact:scope:version:type`. Either the comment or the code isn't quite right, and in this case, I think it's the code.

Reply:
Yes, this ordering is fragile, `explode_coordinates`, `_*_DEP` format strings and `_pom_file_impl` use of exploded coordinates all needs to be in sync, maybe it's time to simplify that by introducing a struct instead of tuple or something like that.

As for the comment, it's kind of correct, refers to input string format ([buildr](https://buildr.apache.org/artifacts.html) flavor), existing branches also change order by putting version into 3rd item of the tuple, while in coordinate string it's always the last part, if present.

Reply:
tried to rewrite this part now